runtime/oti/ContinuationHelpers.hpp

```diff
@@ -187,6 +188,41 @@ class VM_ContinuationHelpers {
 		}
 	}
 
+	static VMINLINE bool
+	isFlatLockAndOwned(j9objectmonitor_t lockWord)
```

Comment:
@babsingh @hangshao0 can you please double check that these methods look right?

Reply:
See https://github.com/eclipse-openj9/openj9/blob/07429785c5dc276272ac3f045c658cfb3cc7162d/runtime/oti/monhelp.h#L28

I think it is safer to always check  `J9_FLATLOCK_OWNER(lockWord) != NULL` before checking `OBJECT_HEADER_LOCK_LEARNING/OBJECT_HEADER_LOCK_RESERVED`. 
This should be equivalent to the existing `J9_LOCK_IS_FLATLOCKED(lock) && J9_FLATLOCK_COUNT(lock) > 0`

Reply:
I will simplify this check